### ISSUE TITLE
Endpoints to support working with contexts

### ISSUE THREAD

**Issue by caro401** [DHARPA-Project/kiara_plugin.service#1]
Please can the REST API expose endpoints for listing, creating and activating (is that the right word?) contexts. This is to support a 'context-switch' component similar to [this streamlit one](https://github.com/DHARPA-Project/kiara_plugin.streamlit/blob/develop/src/kiara_plugin/streamlit/components/context/__init__.py) in the eventual javascript frontend.

**Comment by makkus** [comment]
Probably worth talking about how this would be best implemented. Whether we do it on a session basis where the frontend 'switches' to a different context and from then on the backend will return results depending on that currently active one. 

Or maybe have each context have their own 'root paths', like `<context_name>/data/list` (or whatever endpoint). Or have some sort of context id in the headers. 

Dunno, how would you ideally use this is a frontend dev?

**Comment by caro401** [comment]
I don't know enough about how contexts are intended to work to have an opinion. Do you have design docs/anything written about the user stories etc behind them I could take a look at?

I'd kinda imagined opening/loading something like a project in an IDE (or in tropy), then all the work is done in that project/context until you closed that and opened a different one, but I don't know what would be technically. For tropy, its what sqlite file you're referencing.

The former option sounds like less change required in this plugin? but then would you need to apply a filter or something on every get/post to only target that context?

**Comment by makkus** [comment]
No, sorry, this was one of those things I implemented initially because it made my life developing kiara easier, and then it became clear that it would probably also be useful for users (at least on the command-line) to be able to be able to have several in parallel.

It never really occured to us in the requirements gathering stage, if I remember right. And nobody did any work yet on these kinds of specific UX/UI design questions. Maybe this is a good reason to start something in that space? The backend can probably accommodate whatever works best from a UX perspective, and if not, I'd try to make it so.

Technically, this works similar to tropy it seems, and kiara loads all the information in a specific folder in the user data part of their home directory. If you want to use another context, you need to initialize another KiaraContext object pointing to another folder, which incurs some startup cost, depending on how much data is in there. 
The backend could keep the last context in cache, or garbage collect it, but that probably isn't relevant for the frontend (except for the associated startup cost).
